harness/temporalHelpers.js

```diff
@@ -54,6 +54,101 @@ var TemporalHelpers = {
     { month: 12, monthCode: "M12", daysInMonth: 31 }
   ],
 
+  /*
+   * List of known calendar eras and their possible aliases.
+   *
+   * https://tc39.es/proposal-intl-era-monthcode/#table-eras
+   */
+  CalendarEras: {
+    buddhist: [
+      { era: "buddhist", aliases: ["be"] },
+    ],
+    chinese: [
+      { era: "chinese" },
+    ],
+    coptic: [
+      { era: "coptic" },
+      { era: "coptic-inverse" },
+    ],
+    dangi: [
+      { era: "dangi" },
+    ],
+    ethiopic: [
+      { era: "ethiopic", aliases: ["incar"] },
+      { era: "ethioaa", aliases: ["ethiopic-amete-alem", "mundi"] },
+    ],
+    ethioaa: [
+      { era: "ethioaa", aliases: ["ethiopic-amete-alem", "mundi"] },
+    ],
+    gregory: [
+      { era: "gregory", aliases: ["ce", "ad"] },
+      { era: "gregory-inverse", aliases: ["bc", "bce"] },
+    ],
+    hebrew: [
+      { era: "hebrew", aliases: ["am"] },
+    ],
+    indian: [
+      { era: "indian", aliases: ["saka"] },
+    ],
+    islamic: [
+      { era: "islamic", aliases: ["ah"] },
+    ],
+    "islamic-civil": [
+      { era: "islamic-civil", aliases: ["islamicc", "ah"] },
+    ],
+    "islamic-rgsa": [
+      { era: "islamic-rgsa", aliases: ["ah"] },
+    ],
+    "islamic-tbla": [
+      { era: "islamic-tbla", aliases: ["ah"] },
+    ],
+    "islamic-umalqura": [
+      { era: "islamic-umalqura", aliases: ["ah"] },
+    ],
+    japanese: [
+      { era: "heisei" },
+      { era: "japanese", aliases: ["gregory", "ad", "ce"] },
+      { era: "japanese-inverse", aliases: ["gregory-inverse", "bc", "bce"] },
+      { era: "meiji" },
+      { era: "reiwa" },
+      { era: "showa" },
+      { era: "taisho" },
+    ],
+    persian: [
+      { era: "persian", aliases: ["ap"] },
+    ],
+    roc: [
+      { era: "roc", aliases: ["minguo"] },
+      { era: "roc-inverse", aliases: ["before-roc"] },
+    ],
+  },
+
+  /*
+   * Return the canonical era code.
+   */
+  canonicalizeEraInCalendar(temporalObject, eraName) {
+    if (eraName === undefined) {
+      return undefined;
+    }
+    assert.sameValue(typeof eraName, "string");
+
+    // Don't attempt to canonicalize era names of user-defined calendars.
+    let calendar = temporalObject.getCalendar();
+    if (!(calendar instanceof Temporal.Calendar)) {
+      return eraName;
+    }
+
+    let calendarId = calendar.id;
```

Comment:
Some tests assert that the `id` property of user-defined calendars is only accessed a specific number of times, so we can't directly use `temporalObject.calendarId`, but instead first have to ensure that `calendar` is indeed a `Temporal.Calendar` instance.